### PR TITLE
[Form] 46867 - add more helpful property path accessor exceptions

### PR DESCRIPTION
**Pull request by patrickmaynard**
| Branch?       | 6.3 
| Bug fix?      | no 
| New feature?  | no
| Deprecations? | no
| Tickets       | #46867
| License       | MIT
| Doc PR        | none

Dear reviewers,

This small modification adds more helpful exceptions when properties cannot be accessed/set
using forms. Only one file was modified, so it should be relatively easy to review. Please feel free to leave a comment if you have any questions about what I'm doing here, and thanks for your work as a reviewer on an open-source project!

All the best,

Patrick

### CODE REVIEW

**Comment by xabbuh** [comment]
@patrickmaynard Can you rebase your changes onto the latest `6.2` branch? That will hopefully fix the failures.

**Comment by patrickmaynard** [comment]
> @patrickmaynard Can you rebase your changes onto the latest `6.2` branch? That will hopefully fix the failures.

Currently, when I run `git branch` in my repository, I get this:

```
* 6.2
```

@xabbuh Is there another branch I should be rebasing to? If so, can you provide text of a command that I should be using to do that?

File: src/Symfony/Component/Form/Extension/Core/DataAccessor/PropertyPathAccessor.php [hunk 1]
```diff
@@ -15,7 +15,7 @@
 use Symfony\Component\Form\Exception\AccessException;
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\PropertyAccess\Exception\AccessException as PropertyAccessException;
-use Symfony\Component\PropertyAccess\Exception\NoSuchIndexException;
```
Comment (fabpot) [inline src/Symfony/Component/Form/Extension/Core/DataAccessor/PropertyPathAccessor.php:18]: You should keep this line as this class is referenced in this file.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Form/Extension/Core/DataAccessor/PropertyPathAccessor.php (hunk 1 above)
Comment (patrickmaynard) [reply src/Symfony/Component/Form/Extension/Core/DataAccessor/PropertyPathAccessor.php:18]: Thanks for catching this! I have added the `use` statement for `NoSuchIndexException` back to the file.

**Comment by patrickmaynard** [comment]
@fabpot @xabbuh @yceruto it looks like there is a seemingly unrelated set of unit tests failing with the following information:

```
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    0 => 2023-03-21T13:45:00.000000+0000
-    1 => 2023-03-28T13:45:00.000000+0000
-    2 => 2023-04-04T13:45:00.000000+0000
-    3 => 2023-04-11T13:45:00.000000+0000
-    4 => 2023-04-18T13:45:00.000000+0000
-    5 => 2023-04-25T13:45:00.000000+0000
-    6 => 2023-05-02T13:45:00.000000+0000
-    7 => 2023-05-09T13:45:00.000000+0000
-    8 => 2023-05-16T13:45:00.000000+0000
-    9 => 2023-05-23T13:45:00.000000+0000
-    10 => 2023-05-30T13:45:00.000000+0000
-    11 => 2023-06-06T13:45:00.000000+0000
-    12 => DateTimeImmutable Object (...)
+    0 => 2023-03-28T13:45:00.000000+0000
+    1 => 2023-04-04T13:45:00.000000+0000
+    2 => 2023-04-11T13:45:00.000000+0000
+    3 => 2023-04-18T13:45:00.000000+0000
+    4 => 2023-04-25T13:45:00.000000+0000
+    5 => 2023-05-02T13:45:00.000000+0000
+    6 => 2023-05-09T13:45:00.000000+0000
+    7 => 2023-05-16T13:45:00.000000+0000
+    8 => 2023-05-23T13:45:00.000000+0000
+    9 => 2023-05-30T13:45:00.000000+0000
+    10 => 2023-06-06T13:45:00.000000+0000
+    11 => 2023-06-13T13:45:00.000000+0000
 )
```

Do you know if this is a known generic issue that has a simple fix? It really doesn't seem to have anything to do with the material that I'm changing.

**Comment by xabbuh** [comment]
@patrickmaynard Can you rebase your changes onto the latest `6.3` branch? That should solve the test issue as a bugfix for those has been merged into the Scheduler component recently.

**Comment by patrickmaynard** [comment]
> @patrickmaynard Can you rebase your changes onto the latest `6.3` branch? That should solve the test issue as a bugfix for those has been merged into the Scheduler component recently.

Unfortunately, I ran into some problems while doing this, which corrupted my branch history. Since this is just a one-file change, I will create a new branch and pull request based on 6.3. Thanks!

**Pull request closed without merging**